vmm/src/device_manager.rs

```diff
@@ -1250,12 +1379,12 @@ impl DeviceManager {
         Ok(())
     }
 
-    pub fn io_bus(&self) -> &devices::Bus {
-        &self.io_bus
+    pub fn io_bus(&self) -> &Arc<devices::Bus> {
```

Comment:
I'd rather we kept ownership on DeviceManager rather than moving the ownership to the AddressManager, it feels a bit "weird". This shouldn't cause any problems as it's an Arc<> so should it can just take a clone as part of its contruction